html/js/tree/ISet.js

```diff
@@ -0,0 +1,151 @@
+GTE.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new ISet.
+    * @class
+    * @param {ISet} [parent] Parent ISet. If null, this is root.
+    * @param  {Number} [numOfNodes] Number of nodes of this new information set. If null 1
+    */
+    function ISet(parent, numOfNodes) {
+        this.parent = parent;
+        if (this.parent !== null) this.parent.addChild(this);
+        this.numOfNodes = numOfNodes || 1;
+         // TODO #11
+        var newNodeLevel;
+        if (parent === null) {
+            newNodeLevel = 0;
+        } else {
+            newNodeLevel = parent.node.level + 1;
+        }
+        this.node = new GTE.TREE.Node(this, newNodeLevel);
+        this.moves = [];
+    }
+
+    // TODO #12
+    /**
+    * ToString function
+    */
+    ISet.prototype.toString = function () {
+        return "ISet: " + "parent " + this.parent + "; numOfNodes: " + this.numOfNodes + "; moves: " + this.moves;
+    };
+
+    /**
+    * Function that adds a node to ISet
+    * @param {Node} node Node to add
+    */
+    ISet.prototype.addNode = function (node) {
+        this.nodes.push(node);
+    };
+
+    /**
+    * Function that adds a child ISet to the ISet
+    * @param {ISet} childISet ISet to add
+    */
+    ISet.prototype.addChild = function (childISet) {
+        // Create a move that leads to the child ISet
+        this.moves.push(new GTE.TREE.Move(this, childISet));
+    };
+
+    /**
+    * Function that finds the move that leads to a child ISet
+    * @param {ISet}   childISet ISet to look for
+    * @return {Number}          The index of the move in the list of
+    *                           moves that contains the given iSet 
+    */
+    ISet.prototype.findMoveThatLeadsTo = function (iSet) {
```

Comment:
Take into account that more than one move can lead to the same child ISet, once we introduce ISets with more than one node.

e.g., Here _c_ & _d_ are moves that connect the same parent and children Isets, and same goes for _E_ & _F_
![image](https://cloud.githubusercontent.com/assets/822670/8489869/b080d08a-2120-11e5-8be4-e342faac88f0.png)

A solution that seems promising would be to change this function to findMove*s*ThatLeadTo and return an array of indices. But that wouldn't work either, because more than one ISet can be reached through the same Move from a given parent ISet. e.g. _E_ & _F_ here:

![image](https://cloud.githubusercontent.com/assets/822670/8489906/349f158e-2121-11e5-9231-5bd45d1cb3f2.png)

In summary, a parent ISet and a move don't identify in a unique way a given ISet. We need to think of a different way of representing the data structure that links all ISets together. I will think more about it tomorrow. Also maybe @stengel has some ideas around this. 

(If I didnt explain it properly, please tell me and we can do a quick skype session)